### PR TITLE
Change default player name to use the OS username if available (resolves #2761)

### LINKED ISSUE #2761: Playername doesn't get automatically set
When opening the server screen, the default name is "Player". This could be easily made smarter by defaulting to the current username.

### PR DESCRIPTION
**Pull request by Overv**

### CODE REVIEW

**Comment by LinusU** [comment]
Maybe the default name should be put in a common place instead of hardcoded in two different places?

**Comment by Overv** [comment]
Fixed. The only problem with this is that it will set the default player name before the user has chosen the language, so it will always end up being the English "Player".

**Comment by janisozaur** [comment]
https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by Overv** [comment]
?

File: src/platform/posix.c [hunk 1]
```diff
@@ -876,4 +877,14 @@ datetime64 platform_get_datetime_now_utc()
 	return utcNow;
 }
 
+utf8* platform_get_username() {
+	struct passwd* pw = getpwuid(getuid());
+
+	if (pw) {
+		return pw->pw_name;
+	} else {
+		return language_get_string(STR_PLAYER_DEFAULT_NAME);
```
Comment (IntelOrca) [inline src/platform/posix.c]: I think this should return NULL. ```STR_PLAYER_DEFAULT_NAME``` is specific to multiplayer not getting the current user name from the operating system.

Comment (Overv) [inline src/platform/posix.c]: You're right, fixed it.

**Comment by PFCKrutonium** [comment]
I know I am late, but what about on Windows 8, 8.1 and 10, where your username could be your email address?

**Comment by Overv** [comment]
I'm on Windows 10 and it appears to always use the same name as the folder
in "Users". GetUsername is a legacy function anyway, all of the new formats
(e-mail, full name) need to be retrieved with GetUsernameEx.

**Pull request merged**